### DISCUSSION TITLE
Incorrect Cl Values?

### DISCUSSION THREAD

**Discussion by iabhishek44** [ProjectPhysX/FluidX3D#250] [Q&A]
I made a D2Q9 program on fluidx3d which uses a custom airfoil shape function. In order to validate it, I have been using NASA's turbulence resource page as a reference. However, the results that I have been getting aren't accurate and are off by a big factor. The Reynolds no being used is 6e6, the lbm_u value is 0.5 (i would really like to know how the conversion factor works here? Thanks!)

`units.set_m_kg_s(c, lbm_u, 1.0f, 1.0f, si_u, si_rho);`

    LBM lbm(x_length, y_length, z_length, units.nu_from_Re(Re, c, si_u));
    // ###################################################################################### define geometry ######################################################################################

    const uint Nx=lbm.get_Nx(), Ny=lbm.get_Ny(), Nz=lbm.get_Nz(); parallel_for(lbm.get_N(), [&](ulong n) { uint x=0u, y=0u, z=0u; lbm.coordinates(n, x, y, z);
        if(airfoil_cam(x, y, z, float3(Nx/2u, Ny/2u, Nz/2u), c, t, p, m, aoa)) {lbm.flags[n] = TYPE_S;}
        if(lbm.flags[n] != TYPE_S) lbm.u.x[n] = lbm_u;
        if(x==0u||x==Nx-1u||y==0u||y==Ny-1u) lbm.flags[n] = TYPE_E; // all non periodic
    });

**Comment by CbsTau** [comment]
Hello,

Please can you provide the full setup as it is not easy to advise / debug the model without the full setup you are running.

There are some other discussion threads which deal with determining accurate forces on bodies in FluidX3D:
- https://github.com/ProjectPhysX/FluidX3D/issues/127
- https://github.com/ProjectPhysX/FluidX3D/discussions/58
- https://github.com/ProjectPhysX/FluidX3D/discussions/192

The bottom line seems to be that without the capability to do local mesh refinement around the bodies, and without a proper wall model, the accuracy of force calculation is limited by the overall GPU resources for a mesh of uniform gid size. 

According to Dr Moritz Lehmann in [Issue 127](https://github.com/ProjectPhysX/FluidX3D/issues/127),  "With the resolution you get on a typical desktop GPU, expect drag force to be too large by about a factor 2, due to lack of grid refinement and no wall model."

This mesh resolution issues should be less of a problem in your case if you are doing a 2D (D2Q9) simulation, so you should be able to get closer to the real force values for Cl and Cd.

Regards,
Clinton.
